lib/providers/openstreetmap.js

```diff
@@ -0,0 +1,172 @@
+var _ = require('lodash');
+
+var vumigo = require('vumigo_v02');
+var JsonApi = vumigo.http.api.JsonApi;
+
+var provider = require('./provider');
+var Provider = provider.Provider;
+var AddressResult = provider.AddressResult;
+
+var OpenStreetMap = Provider.extend(function (self, opts) {
+    /**class:OpenStreetMap(opts)
+
+    An address search provider for Open Street Map.
+
+    :param string opts.api_url:
+        The URL of the Open Street Map API. Defaults to
+        'http://open.mapquestapi.com/nominatim/v1/search.php'.
+
+    :param function opts.extract_address_label:
+        Function that extracts a human-friendly label from an Open Street Map
+        API result. See :meth:`OpenStreetMap.extract_address_label` for the
+        function signature and default implementation.
+
+    :param function opts.extract_address_data:
+        Function that extracts data to store on a contact from an Open Street Map
+        API result. See :meth:`OpenStreetMap.extract_address_data` for the
+        function signature and default implementation.
+
+    :param boolean hard_boundary:
+        Whether to limit results to a fixed bounding box. Defaults to `true`.
+
+    :param integer address_limit:
+        Maximum number of search results to return. Defaults to `30`.
+
+    :param array bounding_box:
+        Bounding box to limit search results to. Defaults to
+        `["-180.0", "90.0", "180.0", "-90.0"]`.
```

Comment:
I think we need to clarify this a bit, maybe add link to nominatim docs (`http://open.mapquestapi.com/nominatim/`), or something like this: https://github.com/praekelt/go-jsbox-location/blob/feature/issue-17-add-openstreetmap-lookups/lib/location.js#L311